### ISSUE TITLE
y in core.net

### ISSUE THREAD

**Issue by x2ss** [Vanixxz/MEDAF#4]
Hello, 

Your work is very inspiring and helpful. 

Now, I am trying to write an inference for a single image using a trained model, but encountered a problem. 

The **y** in  [net.py](https://github.com/Vanixxz/MEDAF/blob/main/core/net.py#L196)  is the label and keeps changing. However when doing inference, the label is unknown, and I only input image, and got error: **local variable 'cams' referenced before assignment.** .

Since when doing inference, the label is unknown and needed to be predicted, maybe another net should be used to do inference?

Thank you for your help, best wishes!

This is my code for inference:
**code:**
import os
import torch
from PIL import Image
import numpy as np
import datetime
import torch.nn as nn
import torch.backends.cudnn as cudnn

from core import *
from misc import *
from datasets.osr_loader import CIFAR10_OSR, CIFAR100_OSR, SVHN_OSR, Tiny_ImageNet_OSR

import warnings
warnings.filterwarnings('ignore')

if __name__ == '__main__':
    cudnn.benchmark = True

    #
    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
    print(f"Using device: {device}")

    #
    options = get_config('osr')
    options['img_size'] = 32
    splits = splits_AUROC
    known = splits[options['dataset']][0]
    options['num_known'] = len(known)
    print(f"known: {known}")

    #
    model = get_model(options)

    # 
    checkpoint_path = '/home/cooper/MEDAF/ckpt/osr/cifar10/0814_03:13/epoch_150.pth'  
    checkpoint = torch.load(checkpoint_path, map_location=device)

    # 
    model.load_state_dict(checkpoint['state_dict'])
    model = model.to(device)

    # 
    model.eval()

    # 
    img_path = '/home/cooper/MEDAF/test_data/0_label_3.png'  
    img = Image.open(img_path)
    # 
    img_tensor = torch.tensor(np.array(img)).unsqueeze(0).permute(0, 3, 1, 2).float()  
    img_tensor = img_tensor.to(device)
    print(img_tensor.shape)
    # 
    with torch.no_grad():
        output_dict = model(img_tensor)

    # 
    print("Model output:", output_dict)

**Comment by x2ss** [comment]
I find that for inference，the return_ft should be set to true as https://github.com/Vanixxz/MEDAF/blob/main/core/test.py#L49.

Thanks for sharing the code!
